src/Centreon/Infrastructure/Monitoring/MonitoringRepositoryRDB.php

```diff
@@ -629,6 +695,95 @@ public function findOneService(int $hostId, int $serviceId): ?Service
         return $service;
     }
 
+    /**
+     * @inheritDoc
+     */
+    public function findMultipleServices(array $serviceIds): array
+    {
+        if ($this->hasNotEnoughRightsToContinue()) {
+            return null;
+        }
+
+        $collector = new StatementCollector();
+
+        $accessGroupFilter = $this->isAdmin()
+            ? ' '
+            : ' INNER JOIN `:dbstg`.`centreon_acl` acl
+                  ON acl.host_id = h.host_id
+                  AND acl.service_id = srv.service_id
+                INNER JOIN `:db`.`acl_groups` acg
+                  ON acg.acl_group_id = acl.group_id
+                  AND acg.acl_group_activate = \'1\'
+                  AND acg.acl_group_id IN (' . $this->accessGroupIdToString($this->accessGroups) . ') ';
+
+        $request =
+            'SELECT DISTINCT srv.*,
+              h.host_id AS `host_host_id`, h.name AS `host_name`, h.alias AS `host_alias`, h.instance_id AS `host_poller_id`,
+              srv.state AS `status_code`,
+              CASE
+                WHEN srv.state = 0 THEN "' . ResourceStatus::STATUS_NAME_OK . '"
+                WHEN srv.state = 1 THEN "' . ResourceStatus::STATUS_NAME_WARNING . '"
+                WHEN srv.state = 2 THEN "' . ResourceStatus::STATUS_NAME_CRITICAL . '"
+                WHEN srv.state = 3 THEN "' . ResourceStatus::STATUS_NAME_UNKNOWN . '"
+                WHEN srv.state = 4 THEN "' . ResourceStatus::STATUS_NAME_PENDING . '"
+              END AS `status_name`,
+              CASE
+                WHEN srv.state = 0 THEN ' . ResourceStatus::SEVERITY_OK . '
+                WHEN srv.state = 1 THEN ' . ResourceStatus::SEVERITY_MEDIUM . '
+                WHEN srv.state = 2 THEN ' . ResourceStatus::SEVERITY_HIGH . '
+                WHEN srv.state = 3 THEN ' . ResourceStatus::SEVERITY_LOW . '
+                WHEN srv.state = 4 THEN ' . ResourceStatus::SEVERITY_PENDING . '
+              END AS `status_severity_code`
+            FROM `:dbstg`.services srv
+            LEFT JOIN `:dbstg`.hosts h
+              ON h.host_id = srv.host_id'
+            . $accessGroupFilter
+            . ' WHERE srv.enabled = \'1\'
+              AND h.enabled = \'1\'';
+
+
+        if (is_array($serviceIds)) {
+            $idsListKey = [];
+            foreach ($serviceIds as $x => $hostServiceIds) {
```

Comment:
could you use a better naming than $x ? (maybe $index ?)

Reply:
Sure :+1:

Reply:
done :heavy_check_mark: